.github/workflows/ci.yml

```diff
@@ -231,6 +232,33 @@ jobs:
       - uses: Swatinem/rust-cache@v2
       - run: cargo build --all
 
+  freebsd:
+    timeout-minutes: 10
+    runs-on: ubuntu-24.04
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
```

Comment:
Any particular reason to use hashes instead of version number like we do in all other actions?

Reply:
because of dependency injection attacks